server.js

```diff
@@ -347,6 +347,55 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Rust download and version integration
+camp.route(/^\/crates\/(d|v|dv)\/([A-Za-z0-9_-]+)(?:\/([0-9.]+))?\.(svg|png|gif|jpg|json)$/,
+cache(function (data, match, sendBadge, request) {
+  var mode = match[1];  // d - downloads (total or for version), v - (latest) version, dv - downloads (for latest version)
+  var crate = match[2];  // crate name, e.g. rustc-serialize
+  var version = match[3];  // crate version in semver format, optional, e.g. 0.1.2
+  var format = match[4];
+  var apiUrl = 'https://crates.io/api/v1/crates/' + crate;
+  if (version != null) {
+    apiUrl += '/' + version;
+  }
+  var badgeData = getBadgeData(mode == 'v'? 'version': 'downloads', data);
+  request(apiUrl, { headers: { 'Accept': 'application/json' } }, function (err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(badgeData, format);
+    }
+    try {
+      var data = JSON.parse(buffer);
+      var downloads;
+
+      if (mode == 'd') {
+        downloads = data.crate? data.crate.downloads: data.version.downloads;
+        version = data.version && data.version.num;
+        badgeData.text[1] = metric(downloads) + (version? ' version ' + version: ' total');
+        badgeData.colorscheme = downloadCountColor(downloads);
+
+      } else if (mode = 'dv') {
+        downloads = data.version? data.version.downloads: data.versions[0].downloads;
+        version = data.version && data.version.num;
+        badgeData.text[1] = metric(downloads) + (version? ' version ' + version: ' latest version');
```

Comment:
Technically it's not necessary to shadow version once again, because if the request was successful, and it was for concrete version, the `data.version.num` should be equal to original `version` from the URL, but I left it this way as an additional sanity check. Maybe I'm overprotective here.